app/modules/individuals/models.py

```diff
@@ -231,15 +287,129 @@ def merge_from(self, *source_individuals, sex=None, primary_name=None):
+    # - just a light wrapper to _merge_request_init()
+    def merge_request_from(self, source_individuals, parameters=None):
+        res = self._merge_request_init(source_individuals, parameters)
+        Individual.merge_request_notify([self] + source_individuals, res)
+        return res
+
+    # note: for merge_complete individuals will only contain target individual, but
+    #   request_data should contain key from_individual_ids
+    @classmethod
+    def merge_request_notify(cls, individuals, request_data, notif_type=None):
+        from flask_login import current_user
+        from app.modules.notifications.models import NotificationType
+
+        if not notif_type:
+            notif_type = NotificationType.individual_merge_request
+        owners = {}
+        for indiv in individuals:
+            for enc in indiv.encounters:
+                # we only skip current_user when it is merge_request (merge_complete goes to all users)
+                if not enc.owner or (
+                    notif_type == NotificationType.individual_merge_request
+                    and enc.owner == current_user
+                ):
+                    continue
+                if enc.owner not in owners:
+                    owners[enc.owner] = {'individuals': set(), 'encounters': set()}
+                owners[enc.owner]['individuals'].add(indiv)
+                owners[enc.owner]['encounters'].add(enc)
+        log.debug(
+            f'merge_request_notify() type={notif_type} created owners structure {owners}'
+        )
+        for owner in owners:
+            Individual._merge_request_notify_user(
+                current_user,
+                owner,
+                owners[owner]['individuals'],
+                owners[owner]['encounters'],
+                request_data,
+                notif_type,
+            )
+
+    @classmethod
+    def _merge_request_notify_user(
+        cls, sender, user, individuals, encounters, request_data, notif_type
+    ):
+        from app.modules.notifications.models import (
+            Notification,
+            NotificationBuilder,
+        )
+
+        builder = NotificationBuilder(sender)
+        builder.set_individual_merge_request(individuals, encounters, request_data)
+        notification = Notification.create(notif_type, user, builder)
+        log_msg = (
+            f'merge request: notification {notification} from {sender} re: {individuals}'
+        )
+        AuditLog.audit_log_object(log, user, log_msg)
+
+    @classmethod
+    def merge_request_cancel_task(cls, req_id):
+        current_app.celery.control.revoke(req_id)
+        Individual.merge_request_cleanup(req_id)
+
+    # scrubs notifications etc, once a merge has completed
+    @classmethod
+    def merge_request_cleanup(cls, req_id):
+        log.debug(
+            f'[{req_id}] merge_request_cleanup (notifications, etc) not yet implemented'
+        )
+
+    @classmethod
+    def merge_request_celery_task(
```

Comment:
Celery task code can be tested (by mocking them to circumvent the need for celery) 
See 
https://github.com/WildMeOrg/houston/blob/main/tests/modules/asset_groups/resources/test_create_asset_group.py#L423-L428